### DISCUSSION TITLE
Can I use one theme jar file in many keycloack realms?

### DISCUSSION THREAD

**Discussion by zlatkojosic** [keycloakify/keycloakify#478] [Q&A]
Hi,

I want to have only one theme jar file that support different themes for many keycloak realms. Theme jar will contain artefacts for all themes for all realms. When user come to keycloak from an application we can show right theme due to query parameter in url sent from the application. But we also want to show right theme when user directly go to keycloak. We could achieve that by building jar file for every realm but we want to avoid that. Is it possible in keycloak  realm somehow specify parameter that theme can know what realm it is?

**Comment by garronej** [comment]
Hi @zlatkojosic,  
Yes, there is a build option for this usecase. Look at the theme name option. https://docs.keycloakify.dev/build-options#themename
